src/hotspot/share/nmt/nmtNativeCallStackStorage.hpp

```diff
@@ -40,10 +40,11 @@
 // - Have fast comparisons
 // - Have constant time access
 // We achieve this by using a closed hashtable for finding previously existing NCS:s and referring to them by an index that's smaller than a pointer.
-class NativeCallStackStorage : public CHeapObj<mtNMT> {
+template<template<typename, MEMFLAGS> class ALLOCATOR>
+class NativeCallStackStorageWithAllocator : public CHeapObj<mtNMT> {
```

Comment:
I don't think we need a templatized version of NativeCallStackStorage. Please let's keep it simple and stupid. We will only ever have one form of NativeCallStackStorage.

Should we ever need multiple forms of NativeCallStackStorage simultaneously, we can still templatize.

Reply:
Making this a template also makes it difficult to move implementation over to a cpp file. Something which you really should do, since the implementation here is non-trivial.